configure sql timeouts

cbeck88 opened the pull request:
Brings updates related to SQL timeouts to the release 1.1.2 branch

cbeck88 committed:
configure sql timeouts

these values were suggested to try to improve behavior if we lose
connection to postgres

jgreat reviewed (changes requested):
Timings look good,  Can we get these setting exposed as config values with these setting as default (ideally as env vars, but I'll take cli args, or both).  

We may need to tune and adjust, and it would be nice to not have to cut a release to do that.

cbeck88 commented:
@jgreat let's review this version first, then i'll backport a version of that to 1.1.2, cause it's a larger change to make this configurable everywhere and i'm trying to do it in the cleanest way

https://github.com/mobilecoinfoundation/mobilecoin/pull/1204

cbeck88 committed:
Make sql connection timings configurable (as in master)

This backports part of the sql config PR in mobilecoin repo to
the 1.1.2 release branch

cbeck88 commented:
This is now a backport of the PR merged into master for this, I think it should be ready to go

cbeck88 committed:
fix build
